### PR TITLE
docstring for pad, stride, dilation

### PR DESCRIPTION
**Pull request by saswatpp**
Added Docstrings for expected argument types for pad, stride and dilation in conv layers

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -100,6 +104,10 @@ In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 
 Takes the keyword arguments `pad`, `stride` and `dilation`.
+For Input dimension N, 
+`pad` should be a single Integer or a tuple of with 2*(N-2) parameters indicating
```
Comment (CarloLucibello) [inline src/layers/conv.jl]: of with

File: src/layers/conv.jl [hunk 2]
```diff
@@ -27,6 +27,10 @@ In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 
 Takes the keyword arguments `pad`, `stride` and `dilation`.
+For Input dimension N,
```
Comment (CarloLucibello) [inline src/layers/conv.jl]: input

File: src/layers/conv.jl (hunk 1 above)
Comment (saswatpp) [reply src/layers/conv.jl]: fixed

File: src/layers/conv.jl (hunk 2 above)
Comment (saswatpp) [reply src/layers/conv.jl]: fixed

File: src/layers/conv.jl [hunk 3]
```diff
@@ -27,6 +27,10 @@ In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 
 Takes the keyword arguments `pad`, `stride` and `dilation`.
+For input dimension N, 
+`pad` should be a single Integer or a tuple with 2*(N-2) parameters indicating
```
Comment (CarloLucibello) [inline src/layers/conv.jl]:
```suggestion
`pad` should be a single Integer or a tuple of length 2*(N-2) indicating 
```

File: src/layers/conv.jl [hunk 4]
```diff
@@ -27,6 +27,10 @@ In other words, a 100×100 RGB image would be a `100×100×3×1` array,
 and a batch of 50 would be a `100×100×3×50` array.
 
 Takes the keyword arguments `pad`, `stride` and `dilation`.
```
Comment (CarloLucibello) [inline src/layers/conv.jl]: can you add these keywords to the function signature on the top of this docstring?

File: src/layers/conv.jl (hunk 3 above)
Comment (CarloLucibello) [reply src/layers/conv.jl]: Also, maybe better use something like "for D spatial dimensions, `pad` should be a single Integer or a tuple of length 2*D"

File: src/layers/conv.jl (hunk 4 above)
Comment (bhvieira) [reply src/layers/conv.jl:29]: I think they were there already? in `src/layers/conv.jl` the signature already reads (correctly):
```
    Conv(size, in => out, σ = identity; init = glorot_uniform,
         stride = 1, pad = 0, dilation = 1)
```

File: src/layers/conv.jl [hunk 5]
```diff
@@ -27,6 +27,10 @@ In other words, a 100×100 RGB image would be a `100×100×3×1` array,
...
+For input dimension N, 
+`pad` should be a single Integer or a tuple of length 2*(N-2) indicating
```
Comment (bhvieira) [inline src/layers/conv.jl:31]: Not entirely true, pad can be a `N-2`-D `Tuple` as well, that would apply symmetric padding.

**Comment by bhvieira** [comment]
This PR should close #1077

**Comment by dhairyagandhi96** [comment]
Bump. What's the status on this?

File: src/layers/conv.jl (hunk 4 above)
Comment (CarloLucibello) [reply src/layers/conv.jl:29]: they are not in the docstring's signature:
```
"""
    Conv(size, in=>out)
    Conv(size, in=>out, relu)

Standard convolutional layer. `size` should be a tuple like `(2, 2)`.
```

File: src/layers/conv.jl (hunk 5 above)
Comment (CarloLucibello) [reply src/layers/conv.jl:31]: good catch

File: src/layers/conv.jl (hunk 4 above)
Comment (bhvieira) [reply src/layers/conv.jl:29]: They are though; actually it's the sole signature presented.

Comment (bhvieira) [reply src/layers/conv.jl:29]: I think you are not in `master`, prior to #853 being merged

File: src/layers/conv.jl (hunk 5 above)
Comment (saswatpp) [reply src/layers/conv.jl]: made changes in new commit

**Pull request merged**